src/workerd/io/observer.h

```diff
@@ -127,6 +128,20 @@ class RequestObserver: public kj::Refcounted {
     return nullptr;
   }
 
+  // If the worker is configured to support streaming tail workers, reportTailEvent
+  // will forward the given event on to the collection of streaming tail workers
+  // that are configured with this observer. Otherwise, this is a non-op.
+  virtual void reportTailEvent(IoContext& ioContext, tracing::TailEvent::Event&& event) {
```

Comment:
Q: How can a function be virtual and have an implementation at the same time?

Reply:
That how we allow for default implementations of virtual functions.